webapp/src/views/home/home.vue

```diff
@@ -159,19 +164,23 @@ export default {
     },
   },
   methods: {
-    ...mapActions(['getNewUser', 'getStats']),
+    ...mapActions(['getNewUser', 'getStats', 'setAnonymousDonationDetails']),
     formatWithCommaSeparator,
     handleDonateBtn() {
       this.$bvModal.show('donate-anonymously');
-    }
+    },
   },
   async mounted(){
     if (this.$route.name === 'signup-new-user' && this.$route.params.id && this.$route.params.id.length) {
       await this.getNewUser(this.$route.params.id);
       if (this.message.type !== 'error') this.$bvModal.show('sign-up');
     }
-    else await this.getStats();
-  } 
+    else if (this.$route.name === 'home' && this.$route.query.donate && this.$route.query.name && this.$route.query.email && this.$route.query.phone && this.$route.query.amount) {
+      const { name, email, phone, amount } = this.$route.query;
+      await this.setAnonymousDonationDetails({ name, email, phone, amount: Number(amount) });
+    }
+    await this.getStats();
+  },
```

Comment:
I think we should rename the query option to `showDonate=true` instead of `donate`. Or what do you think? Or maybe the shorter the better so that links we create aren't too long? Or maybe abbreviations like `donate=t&n=John&e=john@mailer.com&p=7232&a=1000`? Does that make a difference? Or we just keep it the way it was?

Reply:
I agree. We should go with something that doesn't take too much space. So I think `donate=t&n=John&e=john@mailer.com&p=7232&a=1000` will do